src/register/SearchAbleDropDown.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Icon } from '@edx/paragon';
+import { ExpandMore, ExpandLess } from '@edx/paragon/icons';
+import onClickOutside from 'react-onclickoutside';
+import PropTypes from 'prop-types';
+import { FormGroup } from '../common-components';
+
+class SearchAbleDropDown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '', icon: ExpandMore };
+  }
+
+  handleClick = () => {
+    const { options } = this.props;
+
+    if (this.state.value === '') {
+      const dropDownItems = options.map(({ code, name }) => (
+        <button type="button" className="dropdown-item" value={name} key={code} onClick={(e) => { this.countryHandler(e); }}>{name} </button>
+      ));
+      this.setState({ dropDownItems, icon: ExpandLess });
+    }
+  }
+
+    handleOnChange = (e) => {
+      const { options } = this.props;
+      const findstr = e.target.value;
+
+      if (findstr.length >= 3) {
```

Comment:
I guess we should start searching right after the user enters the first character.